kollect-core/src/test/kotlin/kollect/InMemoryCacheTest.kt

```diff
@@ -0,0 +1,40 @@
+package kollect
+
+import arrow.core.Tuple2
+import arrow.test.UnitSpec
+import arrow.test.laws.MonoidLaws
+import arrow.test.laws.SemigroupLaws
+import arrow.typeclasses.*
+import io.kotlintest.KTestJUnitRunner
+import io.kotlintest.matchers.shouldNotBe
+import org.junit.runner.RunWith
+
+@RunWith(KTestJUnitRunner::class)
+class InMemoryCacheTest : UnitSpec() {
+
+    val EQ: Eq<InMemoryCache> = object : Eq<InMemoryCache> {
+        override fun eqv(a: InMemoryCache, b: InMemoryCache): Boolean = a.state == b.state
+    }
+
+    private fun dsCache(value: Int): InMemoryCache = InMemoryCache(Tuple2(Tuple2("ds", "key"), value))
+
+    init {
+
+        "instances can be resolved implicitly" {
+            monoid<InMemoryCache>() shouldNotBe null
+        }
+
+
+        testLaws(
+                MonoidLaws.laws(InMemoryCache.monoid(), dsCache(1), EQ),
+                SemigroupLaws.laws(InMemoryCache.monoid(),
```

Comment:
I think the monoid laws already include the semigroup ones

Reply:
It seems not to be the case:
https://github.com/arrow-kt/arrow/blob/c322ce642e16cb8991c1b17c8800b9015eb8b8ae/modules/core/arrow-test/src/main/kotlin/arrow/test/laws/MonoidLaws.kt#L8-L18
https://github.com/arrow-kt/arrow/blob/c322ce642e16cb8991c1b17c8800b9015eb8b8ae/modules/core/arrow-test/src/main/kotlin/arrow/test/laws/SemigroupLaws.kt#L8-L12
Maybe monoid should include associative law in their tests